internal/gengapic/gengapic_test.go

```diff
@@ -585,3 +587,61 @@ func Test_optionsParse(t *testing.T) {
 		}
 	}
 }
+
+func Test_parseRequestHeaders(t *testing.T) {
+	for _, tst := range []struct {
+		name, pattern string
+		want          [][]string
+	}{
+		{"not annotated", "", nil},
+		{"no params", "/no/params", nil},
+		{"not key-value", "/{foo}/{bar}", [][]string{
+			{"{foo", "foo"},
+			{"{bar", "bar"},
+		}},
+		{"key-value pair", "/{foo=blah}/{bar=blah}", [][]string{
+			{"{foo", "foo"},
+			{"{bar", "bar"},
+		}},
+		{"with underscore", "/{foo_foo}/{bar_bar}", [][]string{
+			{"{foo_foo", "foo_foo"},
+			{"{bar_bar", "bar_bar"},
+		}},
+		{"with dot", "/{foo.foo}/{bar.bar}", [][]string{
+			{"{foo.foo", "foo.foo"},
+			{"{bar.bar", "bar.bar"},
+		}},
+		{"numbers in field names", "/{foo123}/{bar456}", [][]string{
+			{"{foo123", "foo123"},
+			{"{bar456", "bar456"},
+		}},
+		{"everything", "/{foo.foo_foo123=blah}/{bar.bar_bar456=blah}", [][]string{
+			{"{foo.foo_foo123", "foo.foo_foo123"},
+			{"{bar.bar_bar456", "bar.bar_bar456"},
+		}},
+	} {
+		m := &descriptor.MethodDescriptorProto{}
+		if tst.pattern != "" {
+			m.Options = &descriptor.MethodOptions{}
+
+			err := setHTTPOption(m.Options, tst.pattern)
+			if err != nil {
+				t.Errorf("parseRequestHeaders(%s): failed to set http annotation: %v", tst.name, err)
+			}
+		}
+
+		if got, err := parseRequestHeaders(m); err != nil {
+			t.Error(err)
+		} else if diff := cmp.Diff(got, tst.want); diff != "" {
```

Comment:
```suggestion
			t.Error(err)
			continue
		}
		if diff := cmp.Diff(got, tst.want); diff != "" {
```

Might make it a little easier to reason about the logical flow.